### PR TITLE
Recovery Page for Harmony in MSC mode

### PR DESCRIPTION
**Pull request by MateuszMudita**
JIRA Reference: [CP-2935](https://appnroll.atlassian.net/browse/CP-2935?atlOrigin=eyJpIjoiZGMxOGRhZDYzODI1NDhmZmE3YmEwMWEwMjBmZTRkZDIiLCJwIjoiaiJ9)

### :memo: Description ️

-

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2935]: https://appnroll.atlassian.net/browse/CP-2935?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/generic-view/ui/src/lib/predefined/recovery-mode/recovery-mode.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  },
+  confirmation: {
+    id: "module.recoveryMode.harmony.confirmation",
+  },
+  button: {
+    id: "module.recoveryMode.harmony.action",
+  },
+})
+
+const RecoveryModeUI: FunctionComponent = () => {
```
Comment (lkowalczyk87) [inline libs/generic-view/ui/src/lib/predefined/recovery-mode/recovery-mode.tsx:48]: import { FunctionComponent } from "Core/core/types/function-component.interface"

File: libs/generic-view/ui/src/lib/shared/warrning-box.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  header: {
+    id: "component.warningBox.headerTitle",
+  },
+})
+
+interface WarningBoxProps {
+  children: React.ReactNode
+}
+
+export const WarrningBox: FunctionComponent<WarningBoxProps> = ({
```
Comment (lkowalczyk87) [inline libs/generic-view/ui/src/lib/shared/warrning-box.tsx:52]: import { FunctionComponent } from "Core/core/types/function-component.interface"

File: libs/core/device-initialization/components/devices-initialization-modal-flows/mudita-harmony-msc-initialization-modal-flow.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    dispatch(
+      setDeviceInitializationStatus(DeviceInitializationStatus.Initialized)
+    )
+
+    history.push(URL_MAIN.recoveryMode)
+    // AUTO DISABLED - fix me if you like :)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  return <></>
```
Comment (mkurczewski) [inline libs/core/device-initialization/components/devices-initialization-modal-flows/mudita-harmony-msc-initialization-modal-flow.tsx:45]: Is this going to be added later?

File: libs/core/__deprecated__/renderer/components/core/icon/icon.component.tsx [hunk 4]
```diff
@@ -22,6 +22,7 @@ export enum IconSize {
   Medium = 2.4,
   Big = 2.8,
   Bigger = 3.2,
+  Biggest = 3.6,
```
Comment (mkurczewski) [inline libs/core/__deprecated__/renderer/components/core/icon/icon.component.tsx:25]: Could you please refactor both names to something like:
`Large`
`ExtraLarge`
?

File: libs/core/device-initialization/components/devices-initialization-modal-flows/mudita-harmony-msc-initialization-modal-flow.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  background: ${({ theme }) => theme.color.grey6};
+`
+
+const ConnectingText = styled.p`
+  font-size: ${({ theme }) => theme.fontSize.headline3};
+  line-height: ${({ theme }) => theme.lineHeight.headline3};
+  color: ${({ theme }) => theme.color.white};
+  font-weight: ${({ theme }) => theme.fontWeight.bold};
+  margin: 2.4rem 0 0;
+`
```
Comment (dkarski) [inline libs/core/device-initialization/components/devices-initialization-modal-flows/mudita-harmony-msc-initialization-modal-flow.tsx:60]: to remove?
```suggestion
```

File: libs/core/device-initialization/components/devices-initialization-modal-flows/mudita-harmony-msc-initialization-modal-flow.tsx (hunk 3 above)
Comment (MateuszMudita) [reply libs/core/device-initialization/components/devices-initialization-modal-flows/mudita-harmony-msc-initialization-modal-flow.tsx:45]: No, it just should return empty component.

Comment (mkurczewski) [reply libs/core/device-initialization/components/devices-initialization-modal-flows/mudita-harmony-msc-initialization-modal-flow.tsx:31]: Ah, ok. In that case just a 
```suggestion
  return null
```
should be enough.

**Pull request merged**